### ISSUE TITLE
ESM configs can't load

### ISSUE THREAD

**Issue by the-spyke** [babel/babel-loader#824]
## **BEFORE YOU SUBMIT** please read the following:

**I'm submitting a bug report**

**Node Version:**
13.8

**Webpack Version:**
4.41

**Babel Core Version**:
7.8

**Babel Loader Version**:
8.0.6

**Please tell us about your environment:**
Linux

**Current behavior:**

In case of using `type=module` in `package.json` or `.mjs` extension for ESM configuration file Babel throws an error:

```text
babel.config.js: Error while loading config - You appear to be using a native ECMAScript module configuration file, which is only supported when running Babel asynchronously.
```

**Expected/desired behavior:**

ESM configs are supporter by Babel and should load/work.

This [line in index.js](https://github.com/babel/babel-loader/blob/master/src/index.js#L161) uses synchronous `loadPartialConfig` which should be `loadPartialConfigAsync` instead and `await` (parent function is async already).

**Comment by the-spyke** [comment]
I was trying to fix this, but `babel-loader` loading `.mjs` config files with `import()` inside of `@babel/core` fails with

```
TypeError [ERR_VM_DYNAMIC_IMPORT_CALLBACK_MISSING]: A dynamic import callback was not specified
```

Same time `@babel/cli` reads such config files fine. Didn't look further.

**Comment by nicolo-ribaudo** [comment]
Are you using Jest? I have seen that error with Jest because it doesn't support native `import()` yet.

**Comment by the-spyke** [comment]
@nicolo-ribaudo I’m using Webpack CLI.

**Comment by nicolo-ribaudo** [comment]
I have investigated a bit, and it seems that it is a webpack problem.
If you use this `webpack.config.js` (which doesn't even load Babel):
```js
const path = require("path");

import("./test.mjs").then(console.log);

module.exports = {
  entry: './index.js',
  output: {
    path: path.resolve(__dirname, 'dist'),
    filename: 'bundle.js'
  }
}
```
```js
// test.mjs
export default 1;
```

And run webpack, you'll get the same error:
```
TypeError [ERR_VM_DYNAMIC_IMPORT_CALLBACK_MISSING]: A dynamic import callback was not specified.
```

This usually happens when using Node's `vm` module without properly handling dynamic imports. I'd report this to webpack, since the `import()` in my example is unuseful but it shouldn't throw. If you run `node webpack.config.js`, it doesn't throw.

**Comment by the-spyke** [comment]
@nicolo-ribaudo Yeah, haven't time to check this yesterday.

I want to fix this issue, but the `loadPartialConfigAsync` was added only in `core@7.8.0`. Depending on `7.8.0` is a breaking change. It could use `sync/async` depending on environment instead, but users will have no clue why their `.mjs` config files weren't found. Maybe post a warning in case of `core<7.8.0`?

**Comment by nicolo-ribaudo** [comment]
> I want to fix this issue, but the loadPartialConfigAsync was added only in core@7.8.0. Depending on 7.8.0 is a breaking change.

Right! I think that this would be enough:
```js
await (babel.loadPartialConfigAsync || babel.loadPartialConfig)(args);
```

> but users will have no clue why their .mjs config files weren't found.

Isn't this the same for any feature? e.g. Feature X is introduced in version 7.6, and it isn't available in version 7.5.

**Comment by franciscofabian** [comment]
I have the same issue, is there any timeline on the fix?

**Comment by yanickrochon** [comment]
I came back to an old project after months without updates and, after updating the dependencies, all my tests can't run anymore. All I get is this error 


> [BABEL]: You appear to be using a native ECMAScript module preset, which is only supported when running Babel asynchronously.

I have tried converting `.babelrc` to `babel.config.js` and then `babel.config.cjs`, then also tried JSON with `babel.config.json`, the problem is unchanged.

I am using `jest@20.1.3` with `babel-jest@20.1.3`, and `@babel/core@7.18.10` with `@babel/present-env@7.18.10`.

What's the resolution for this?
